File: src/test/java/de/fraunhofer/aisec/crymlin/CTest.java

package de.fraunhofer.aisec.crymlin;

import org.junit.jupiter.api.Test;

class CTest extends AbstractMarkTest {

	@Test
	void testDefinition() throws Exception {
		var findings = performTest("c/def.c", "c/s.mark");
		containsFindings(findings, "line 14: Rule some_s verified");
	}

	@Test
	void testDeclAssign() throws Exception {
		var findings = performTest("c/declassign.c", "c/s.mark");
		containsFindings(findings, "line 16: Rule some_s verified");
	}
}

File: src/test/java/de/fraunhofer/aisec/codyze/legacy/crymlin/CTest.kt
package de.fraunhofer.aisec.codyze.legacy.crymlin

import org.junit.jupiter.api.Test

internal class CTest : AbstractMarkTest() {
    @Test
    @Throws(Exception::class)
    fun testDefinition() {
        val findings = performTest("legacy/c/def.c", "legacy/c/s.mark")
        containsFindings(findings, "line 14: Rule some_s verified")
    }

    @Test
    @Throws(Exception::class)
    fun testDeclAssign() {
        val findings = performTest("legacy/c/declassign.c", "legacy/c/s.mark")
        containsFindings(findings, "line 16: Rule some_s verified")
    }
}
